### ISSUE TITLE
Specification or implementation error

### ISSUE THREAD

**Issue by tectiv3** [standardfile/standardfile.github.io#3]
Hello!
I'm trying to implement Standard File protocol in Go and wanted to report that protocol specification says you should return this response on sync request:

`json
{"retrieved_items" : [], "saved_items" : [], "unsaved_items" : [], "sync_token" : ""}
`
but in reality standard notes clients expect this:
`json
{"retrieved_items" : [], "saved_items" : [], "unsaved" : [], "sync_token" : ""}
`

**unsaved** instead of **unsaved_items**

Minor mistake, but iOS app silently crashes and desktop client put sync error in console.

Thank you.

**Comment by mobitar** [comment]
Oh wow good catch. I'll have to either update the clients or update the spec. I think it's better to keep it consistent and use `unsaved_items`. Are you developing the Go engine open source? Would love to see your progress.

**Comment by mobitar** [comment]
Nevermind, [found it](https://github.com/tectiv3/standardfile). Awesome stuff! Seems pretty much built out. Is it fully functional?

**Comment by tectiv3** [comment]
Yeah, almost finished. Needs some refactoring and password change functionality.

**Comment by tectiv3** [comment]
One more:
### Updates a user's password.
#### Documentation
> PATCH auth
> Params: email, password, password_confirmation, current_password

#### Actual client
> POST auth/change_pw
> Params: 
`{"new_password":"","pw_func":"pbkdf2","pw_alg":"sha512","pw_key_size":512,"pw_cost":5000,"pw_salt":"","pw_nonce":""}`

**Comment by tectiv3** [comment]
And another one:

iOS app tries to sign in using:
`POST /api/auth/sign_in.json`
instead of 
`/api/auth/sign_in`

Also it crashes a lot :)

**Comment by mobitar** [comment]
Good finds, thanks.

The iOS app crashes a lot? Under what circumstances? Just regular operation?

**Comment by tectiv3** [comment]
No, not just regular operation, but when trying to consume my API :) Where desktop app would do just fine - iOS crashes. I figured it was mostly on json serialisation stage.
But I think I got it, everything is working smoothly now.

**Comment by mobitar** [comment]
Btw I've added a link to your implementation to https://standardfile.org. Hope that's ok with you.

**Comment by tectiv3** [comment]
Of course, thank you.
